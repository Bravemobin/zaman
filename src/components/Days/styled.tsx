import styled from "../../theme";

export const DaysBody = styled("div")`
  max-width: 310px;
  max-height: 85%;
  position: relative;
  overflow: auto;
  border-radius: ${8 / 16}rem;
  background-color: ${props => props.theme.backColor};
  & * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    user-select: none;
  }
`;

export const DaysHead = styled("div")`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: ${24 / 16}rem;
  background-color: ${props => props.theme.headBackColor};
`;
export const HeadTitle = styled("h4")`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding: 0 ${8 / 16}rem;
  font-size: 1.618rem;
  text-align: center;
  color: ${props => props.theme.headTitleColor};

  svg {
    fill: ${props => props.theme.headTitleColor};
  }
`;

export const HeadRange = styled("h3")`
  margin-top: ${24 / 16}rem;
  margin-bottom: 1rem;
  font-size: 1.618rem;
  border-radius: ${20 / 16}rem;
  padding: 0 ${8 / 16}rem;
  background-color: ${props => props.theme.headRangeBackColor};
  color: ${props => props.theme.headRangeColor};
`;
export const Table = styled("table")<{ timePicker: boolean }>`
  width: 100%;
  font-size: 1rem;
  border-collapse: separate;
  border-spacing: 0 0.5rem;
  padding: ${props => (props.timePicker ? `${8 / 16}rem` : 0)};

  th {
    font-size: 1rem;
    font-weight: 300;
    color: ${props => props.theme.weekDaysColor};
  }
`;

export const ButtonsDiv = styled("div")`
  margin-top: ${16 / 16}rem;
  margin-bottom: ${16 / 16}rem;

  button {
    border: none;
    margin: 0;
    padding: ${8 / 16}rem ${24 / 16}rem;
    width: auto;
    min-width: 80px;
    overflow: visible;
    font: inherit;
    line-height: normal;
    appearance: none;
    outline: 0;
    border-radius: ${4 / 16}rem;
    cursor: pointer;
    &::-moz-focus-inner {
      border: 0;
      padding: 0;
      outline: 0;
    }
  }
  .rdp__button--cancel {
    margin-right: 1rem;
    color: ${props => props.theme.cancelColor};
    background-color: ${props => props.theme.cancelBackColor};
    transition: background-color 0.2s ease;
    &:hover,
    &:focus {
      color: ${props => props.theme.cancelHoverColor};
      background-color: ${props => props.theme.cancelHoverBackColor};
    }
  }
  .rdp__button--submit {
    margin-right: 1rem;
    color: ${props => props.theme.submitColor};
    background-color: ${props => props.theme.submitBackColor};
    transition: background-color 0.2s ease;
    &:hover,
    &:focus {
      color: ${props => props.theme.submitHoverColor};
      background-color: ${props => props.theme.submitHoverBackColor};
    }
  }
`;

export const ChangeViewButton = styled("button")`
  min-width: 40px !important;
  min-height: 40px;
  float: left;
  margin-left: 16px !important;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: background-color 0.2s ease;
  background-color: ${props => props.theme.changeViewButtonBackColor};
  svg {
    fill: ${props => props.theme.changeViewButtonColor};
  }
  &:hover {
    background-color: ${props => props.theme.changeViewButtonHoverBackColor};
    svg {
      fill: ${props => props.theme.changeViewButtonHoverColor};
    }
  }
`;
