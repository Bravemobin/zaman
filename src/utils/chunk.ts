/**
 * Its just a copy from lodash.
 * I used it in this way because package has an issue with rollup
 */

/**
 * Creates a slice of `array` from `start` up to, but not including, `end`.
 *
 * **Note:** This method is used instead of
 * [`Array#slice`](https://mdn.io/Array/slice) to ensure dense arrays are
 * returned.
 *
 * @since 3.0.0
 * @category Array
 * @param {Array} array The array to slice.
 * @param {number} [start=0] The start position. A negative index will be treated as an offset from the end.
 * @param {number} [end=array.length] The end position. A negative index will be treated as an offset from the end.
 * @returns {Array} Returns the slice of `array`.
 * @example
 *
 * var array = [1, 2, 3, 4]
 *
 * _.slice(array, 2)
 * // => [3, 4]
 */
function slice(array: [], start: number, end: number) {
  let length = array == null ? 0 : array.length;
  if (!length) {
    return [];
  }
  start = start == null ? 0 : start;
  end = end === undefined ? length : end;

  if (start < 0) {
    start = -start > length ? 0 : length + start;
  }
  end = end > length ? length : end;
  if (end < 0) {
    end += length;
  }
  // @ts-ignore
  // tslint:disable-next-line:no-bitwise
  length = start > end ? 0 : (end - start) >>> 0;
  // tslint:disable-next-line:no-bitwise
  start >>>= 0;

  let index = -1;
  const result = new Array(length);
  while (++index < length) {
    result[index] = array[index + start];
  }
  return result;
}

/**
 * Creates an array of elements split into groups the length of `size`.
 * If `array` can't be split evenly, the final chunk will be the remaining
 * elements.
 *
 * @since 3.0.0
 * @category Array
 * @param {Array} array The array to process.
 * @param {number} [size=1] The length of each chunk
 * @returns {Array} Returns the new array of chunks.
 * @example
 *
 * chunk(['a', 'b', 'c', 'd'], 2)
 * // => [['a', 'b'], ['c', 'd']]
 *
 * chunk(['a', 'b', 'c', 'd'], 3)
 * // => [['a', 'b', 'c'], ['d']]
 */
export function chunk(array, size) {
  size = Math.max(size, 0);
  const length = array == null ? 0 : array.length;
  if (!length || size < 1) {
    return [];
  }
  let index = 0;
  let resIndex = 0;
  const result = new Array(Math.ceil(length / size));

  while (index < length) {
    result[resIndex++] = slice(array, index, (index += size));
  }
  return result;
}
