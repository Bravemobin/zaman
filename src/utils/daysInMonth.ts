import { Dayjs } from "dayjs";
import dayjs from "dayjs";
import { fa } from "./index";

export interface IDays {
  day: string;
  // utc: string;
  faDate: string;
  disable: boolean;
}

export interface IDaysInMonth {
  days: IDays[];
  monthName: string;
  month: number;
  today?: string;
}
const checkDateMonth = (date, current) =>
  current.calendar("jalali").month() < date.calendar("jalali").month();
const checkCurrentMonth = (date: Dayjs) =>
  dayjs()
    .calendar("jalali")
    .format("YYYY/MM") === date.calendar("jalali").format("YYYY/MM");

export const daysInMonth = (date: Dayjs): IDaysInMonth => {
  const days: IDays[] = [];
  const monthName = `${date
    .calendar("jalali")
    .locale("fa")
    .format("MMMM")} ${fa(date.calendar("jalali").format("YYYY"))}`;

  const month = +date
    .calendar("jalali")
    .locale("fa")
    .format("M");

  let firstDayOfWeek = date.calendar("jalali").startOf("month");
  const lastDayOfWeek = date.calendar("jalali").endOf("month");
  const today = checkCurrentMonth(date)
    ? { today: date.calendar("jalali").format("DD") }
    : null;

  firstDayOfWeek = firstDayOfWeek.subtract(
    (firstDayOfWeek.day() + 1) % 7,
    "day",
  );

  while (firstDayOfWeek.isBefore(lastDayOfWeek)) {
    days.push({
      day: firstDayOfWeek.calendar("jalali").format("DD"),
      // utc: new Date(firstDayOfWeek.clone().format("YYYY/M/DD")).toUTCString(),
      faDate: firstDayOfWeek.calendar("jalali").format("YYYY/MM/DD"),
      disable: checkDateMonth(date, firstDayOfWeek),
    });
    firstDayOfWeek = firstDayOfWeek.add(1, "day");
  }

  // tslint:disable-next-line:no-console
  // console.log("days ", { monthName, month });
  return { monthName, month, days, ...today };
};
