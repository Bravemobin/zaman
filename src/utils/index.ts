import * as moment from "jalali-moment";

export const fa = n =>
  Number(n).toLocaleString("fa", {
    useGrouping: false,
  });

// prettier-ignore
export const inputFaDateMask = [/[0-1]/,/[0-4]/,/[0-9]/,/[0-9]/, '/', /[0-1]/, /[0-9]/, '/', /[0-3]/, /[0-9]/];
// prettier-ignore
export const inputFaDateWithTimeMask = [/[0-1]/,/[0-4]/,/[0-9]/,/[0-9]/, '/', /[0-1]/, /[0-9]/, '/', /[0-3]/, /[0-9]/, ' ','-',' ', /[0-2]/,/[0-9]/, ':', /[0-5]/,/[0-9]/];
// week days name
export const weekDayNames = ["ش", "ی", "د", "س", "چ", "پ", "ج"];
// format jalali string date into moment
export const formatJalaliDate = date => {
  const formattedDate = moment(`${date}`, "jYYYY/jMM/jDD");
  if (formattedDate.isValid()) {
    return formattedDate;
  }
  return null;
};
