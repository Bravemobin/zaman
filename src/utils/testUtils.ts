import { ITheme } from "../theme";

const days = `[{"day":"01","utc":"Fri, 21 Dec 2018 20:30:00 GMT","faDate":"1397/10/01","disable":false},{"day":"02","utc":"Sat, 22 Dec 2018 20:30:00 GMT","faDate":"1397/10/02","disable":false},{"day":"03","utc":"Sun, 23 Dec 2018 20:30:00 GMT","faDate":"1397/10/03","disable":false},{"day":"04","utc":"Mon, 24 Dec 2018 20:30:00 GMT","faDate":"1397/10/04","disable":false},{"day":"05","utc":"Tue, 25 Dec 2018 20:30:00 GMT","faDate":"1397/10/05","disable":false},{"day":"06","utc":"Wed, 26 Dec 2018 20:30:00 GMT","faDate":"1397/10/06","disable":false},{"day":"07","utc":"Thu, 27 Dec 2018 20:30:00 GMT","faDate":"1397/10/07","disable":false},{"day":"08","utc":"Fri, 28 Dec 2018 20:30:00 GMT","faDate":"1397/10/08","disable":false},{"day":"09","utc":"Sat, 29 Dec 2018 20:30:00 GMT","faDate":"1397/10/09","disable":false},{"day":"10","utc":"Sun, 30 Dec 2018 20:30:00 GMT","faDate":"1397/10/10","disable":false},{"day":"11","utc":"Mon, 31 Dec 2018 20:30:00 GMT","faDate":"1397/10/11","disable":false},{"day":"12","utc":"Tue, 01 Jan 2019 20:30:00 GMT","faDate":"1397/10/12","disable":false},{"day":"13","utc":"Wed, 02 Jan 2019 20:30:00 GMT","faDate":"1397/10/13","disable":false},{"day":"14","utc":"Thu, 03 Jan 2019 20:30:00 GMT","faDate":"1397/10/14","disable":false},{"day":"15","utc":"Fri, 04 Jan 2019 20:30:00 GMT","faDate":"1397/10/15","disable":false},{"day":"16","utc":"Sat, 05 Jan 2019 20:30:00 GMT","faDate":"1397/10/16","disable":false},{"day":"17","utc":"Sun, 06 Jan 2019 20:30:00 GMT","faDate":"1397/10/17","disable":false},{"day":"18","utc":"Mon, 07 Jan 2019 20:30:00 GMT","faDate":"1397/10/18","disable":false},{"day":"19","utc":"Tue, 08 Jan 2019 20:30:00 GMT","faDate":"1397/10/19","disable":false},{"day":"20","utc":"Wed, 09 Jan 2019 20:30:00 GMT","faDate":"1397/10/20","disable":false},{"day":"21","utc":"Thu, 10 Jan 2019 20:30:00 GMT","faDate":"1397/10/21","disable":false},{"day":"22","utc":"Fri, 11 Jan 2019 20:30:00 GMT","faDate":"1397/10/22","disable":false},{"day":"23","utc":"Sat, 12 Jan 2019 20:30:00 GMT","faDate":"1397/10/23","disable":false},{"day":"24","utc":"Sun, 13 Jan 2019 20:30:00 GMT","faDate":"1397/10/24","disable":false},{"day":"25","utc":"Mon, 14 Jan 2019 20:30:00 GMT","faDate":"1397/10/25","disable":false},{"day":"26","utc":"Tue, 15 Jan 2019 20:30:00 GMT","faDate":"1397/10/26","disable":false},{"day":"27","utc":"Wed, 16 Jan 2019 20:30:00 GMT","faDate":"1397/10/27","disable":false},{"day":"28","utc":"Thu, 17 Jan 2019 20:30:00 GMT","faDate":"1397/10/28","disable":false},{"day":"29","utc":"Fri, 18 Jan 2019 20:30:00 GMT","faDate":"1397/10/29","disable":false},{"day":"30","utc":"Sat, 19 Jan 2019 20:30:00 GMT","faDate":"1397/10/30","disable":false}]`;
export const mockDays = JSON.parse(days);

export const theme: ITheme = {
  // head
  headBackColor: "#FFFFFF",
  headTitleColor: "#000",
  headArrowColor: "#000",
  headRangeBackColor: "#D6D6D6",
  headRangeColor: "#000",

  // weekdays color
  weekDaysColor: "#3F3F3F",

  // days
  daysColor: "#000",
  daysBackColor: "#FFFFFF",
  holidaysColor: "#F50057",
  holidaysBackColor: "#FFFFFF",

  // start end
  startEndRangeColor: "#000",
  startRangeBackColor: "#D6D6D6",
  endRangeBackColor: "#D6D6D6",
  continueRangeBackColor: "#f3f3f3",
};

export const rangeHelperMock = {
  "1397/10/04": { status: "startRange" },
  "1397/10/05": { status: "continueRange" },
  "1397/10/06": { status: "continueRange" },
  "1397/10/07": { status: "continueRange" },
  "1397/10/08": { status: "endRange" },
};
