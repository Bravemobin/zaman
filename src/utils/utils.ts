export const fa = n =>
  Number(n).toLocaleString("fa", {
    useGrouping: false,
  });
