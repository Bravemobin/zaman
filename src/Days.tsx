import * as React from "react";
import styled from "styled-components";
import { ThemeProvider } from "styled-components";
import { IDays } from "./utils";
import { defaultTheme } from "./theme";
import { IRangeDays } from "./types";

const DaysHead = styled.div`
  background-color: ${props => props.theme.headBackColor};
`;

const HeadTitle = styled.h4`
  color: ${props => props.theme.headTitleColor};
`;

const HeadRange = styled.h3`
  background-color: ${props => props.theme.headRangeBackColor};
  color: ${props => props.theme.headRangeColor};
`;

export const Day = styled.td`
	${props =>
    !props.startEndRange &&
    !props.holiday &&
    `color: ${props.theme.daysColor};
     background-color: ${props.theme.daysBackColor}`}
  ${props =>
    props.holiday &&
    `color: ${props.theme.holidaysColor};
     background-color: ${props.theme.holidaysBackColor}`}
	${props =>
    props.startEndRange &&
    `color: ${props.theme.startEndDayColor};
     background-color: ${props.theme.startEndDayBackColor}`}
`;

export interface IDaysProps {
  days: IDays[];
  theme?: any;
  rangeDays: IRangeDays;
  daysEvent: any;
}

const boolAttr = (arg: boolean) => {
  if (arg) {
    return {
      "data-disable": arg,
    };
  }
  return null;
};

export class Days extends React.PureComponent<IDaysProps> {
  public static defaultProps: Partial<IDaysProps> = {
    theme: defaultTheme,
  };
  public render(): React.ReactNode {
    const { days, theme, rangeDays, daysEvent } = this.props;
    if (!days.length) {
      return null;
    }
    return (
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <DaysHead data-testid="days-head">
            <HeadTitle data-testid="days-head-title" />
            <HeadRange data-testid="days-head-range" />
          </DaysHead>
          <table>
            <thead />
            <tbody>
              <tr data-testid="days">
                {days.map((day: IDays, id) => (
                  <Day
                    key={`rdp-days-${id}`}
                    data-testid={`day-${id + 1}`}
                    data-fadate={`${day.faDate}`}
                    // data-disable={`${day.disable}`}
                    startEndRange={rangeDays ? rangeDays[day.faDate] : false}
                    {...daysEvent()}
                    {...boolAttr(day.disable)}
                  >
                    {day.day}
                  </Day>
                ))}
              </tr>
            </tbody>
          </table>
        </React.Fragment>
      </ThemeProvider>
    );
  }
}
