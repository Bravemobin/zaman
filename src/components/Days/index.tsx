import * as React from "react";
import { ThemeProvider } from "../../theme";
import { TimePicker } from "../TimePicker";
import { IDaysProps } from "./types";
import { DaysHead } from "../DaysHead";
import { Day } from "../Day";
import {
  ButtonsDiv,
  ChangeViewButton,
  DaysBody,
  Table,
  TimeDays,
} from "./styled";
import { chunk } from "../../utils/chunk";
import { IDays } from "../../utils/daysInMonth";
import { fa, weekDayNames } from "../../utils";

const boolDataset = (arg: boolean) => {
  if (arg) {
    return {
      "data-disable": arg,
    };
  }
  return null;
};

export class Days extends React.PureComponent<IDaysProps> {
  public static defaultProps: Partial<IDaysProps> = {
    monthName: "",
    holiday: [],
    daysEventListeners: () => null,
    timePicker: false,
    timePickerView: false,
  };
  public render(): React.ReactNode {
    const {
      days,
      theme,
      rangeDays,
      daysEventListeners,
      selectedPickerStatus,
      ArrowLeft,
      ArrowRight,
      ClockIcon,
      DateIcon,
      monthName,
      increaseMonth,
      decreaseMonth,
      isSelecting,
      isRenderingButtons,
      onCancelButton,
      onSubmitButton,
      selectedDay,
      timePicker,
      timePickerView,
      toggleView,
      hour,
      minute,
      changeHour,
      changeMinute,
    } = this.props;
    if (!days.length) {
      return null;
    }
    const weeks = chunk(days, 7);
    return (
      <ThemeProvider theme={theme}>
        <DaysBody>
          <DaysHead
            monthName={monthName}
            datePickerStatus={selectedPickerStatus}
            ArrowLeft={ArrowLeft}
            ArrowRight={ArrowRight}
            increaseMonth={increaseMonth}
            decreaseMonth={decreaseMonth}
            hour={hour}
            minute={minute}
            timePickerView={timePickerView}
          />
          {timePicker && timePickerView ? (
            <TimeDays data-testid="dp__timePicker">
              <TimePicker
                hour={hour}
                minute={minute}
                changeHour={changeHour}
                changeMinute={changeMinute}
              />
            </TimeDays>
          ) : (
            <React.Fragment>
              <Table data-testid="table-days" timePicker={timePicker}>
                <thead>
                  <tr>
                    {weekDayNames.map((name, idx) => (
                      <th key={`weekDayName-${idx}`}>{name}</th>
                    ))}
                  </tr>
                </thead>
                <tbody>
                  {weeks.map((week, idx) => (
                    <tr data-testid="days" key={`rdp-weeks-${idx}`}>
                      {week.map((day: IDays, id) => (
                        <Day
                          key={`rdp-days-${id}`}
                          data-testid={`day-${idx * 7 + id + 1}`}
                          data-fadate={`${day.faDate}`}
                          daysEvent={daysEventListeners}
                          theme={theme}
                          startEndRange={rangeDays && rangeDays[day.faDate]}
                          isSelecting={isSelecting}
                          selectedDay={selectedDay === day.faDate}
                          holiday={this.props.holiday.filter(
                            holiday => holiday === id,
                          )}
                          {...boolDataset(day.disable)}
                        >
                          {!day.disable ? fa(day.day) : null}
                        </Day>
                      ))}
                    </tr>
                  ))}
                </tbody>
              </Table>
            </React.Fragment>
          )}
          {isRenderingButtons && (
            <ButtonsDiv className="rdp__buttons" data-testid="rdp__buttons">
              <button
                data-testid="submit-button"
                className="rdp__button--submit"
                onClick={onSubmitButton}
              >
                تایید
              </button>
              <button
                data-testid="cancel-button"
                className="rdp__button--cancel"
                onClick={onCancelButton}
              >
                لغو
              </button>
              {timePicker && (
                <ChangeViewButton
                  onClick={toggleView}
                  data-testid="toggle-view"
                >
                  {timePickerView ? <DateIcon /> : <ClockIcon />}
                </ChangeViewButton>
              )}
            </ButtonsDiv>
          )}
        </DaysBody>
      </ThemeProvider>
    );
  }
}
