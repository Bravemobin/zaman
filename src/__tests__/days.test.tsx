import * as React from "react";
import { cleanup, render } from "react-testing-library";
import { Days, Day } from "../Days";
import "jest-styled-components";
import { mockDays, rangeHelperMock, theme } from "../utils/testUtils";
const mockDaysEvent = jest.fn();

describe("theme test ", () => {
  afterEach(cleanup);
  test("Head test ", () => {
    const { getByTestId } = render(
      <Days theme={theme} days={mockDays} daysEvent={mockDaysEvent} />,
    );
    const daysHead = getByTestId("days-head");
    const daysHeadTitle = getByTestId("days-head-title");
    const daysHeadRange = getByTestId("days-head-range");

    expect(daysHead).toHaveStyleRule("background-color", theme.headBackColor);
    expect(daysHeadTitle).toHaveStyleRule("color", theme.headTitleColor);
    expect(daysHeadRange).toHaveStyleRule(
      "background-color",
      theme.headRangeBackColor,
    );
    expect(daysHeadRange).toHaveStyleRule("color", theme.headRangeColor);
  });

  test("day test ", () => {
    const table = document.createElement("tr");
    const { container } = render(<Day theme={theme} days={mockDays} />, {
      container: table,
    });
    expect(container.firstChild).toHaveStyleRule(
      "background-color",
      theme.daysBackColor,
    );
    expect(container.firstChild).toHaveStyleRule("color", theme.daysColor);
  });

  test("holiday test ", () => {
    const table = document.createElement("tr");
    const { container } = render(
      <Day theme={theme} days={mockDays} holiday />,
      {
        container: table,
      },
    );
    expect(container.firstChild).toHaveStyleRule(
      "background-color",
      theme.holidaysBackColor,
    );
    expect(container.firstChild).toHaveStyleRule("color", theme.holidaysColor);
  });

  test("startEndRange test ", () => {
    const table = document.createElement("tr");
    const { container } = render(
      <Day days={mockDays} theme={theme} startEndRange />,
      {
        container: table,
      },
    );
    expect(container.firstChild).toHaveStyleRule(
      "background-color",
      theme.startEndDayBackColor,
    );
    expect(container.firstChild).toHaveStyleRule(
      "color",
      theme.startEndDayColor,
    );
  });
});

describe("day test ", () => {
  afterEach(cleanup);

  test("when day is empty object ", () => {
    const { container } = render(<Days days={[]} daysEvent={mockDaysEvent} />);
    expect(container.textContent).toBe("");
  });

  test("day ", () => {
    const { getByTestId } = render(
      <Days days={mockDays} daysEvent={mockDaysEvent} />,
    );
    const container = getByTestId("days");
    expect(container.innerHTML).toMatchSnapshot();
  });
});

describe("day withRangeDays test ", () => {
  afterEach(cleanup);

  const { getByTestId } = render(
    <Days
      days={mockDays}
      rangeDays={rangeHelperMock}
      daysEvent={mockDaysEvent}
    />,
  );
  const day = getByTestId("day-4");
  expect(day).toHaveStyleRule("background-color", theme.startEndDayBackColor);
});
