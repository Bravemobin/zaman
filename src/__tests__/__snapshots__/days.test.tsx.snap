// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`day test  day  1`] = `"<td data-testid=\\"day-1\\" data-fadate=\\"1397/10/01\\" class=\\"sc-bdVaJa hyzvNA\\">01</td><td data-testid=\\"day-2\\" data-fadate=\\"1397/10/02\\" class=\\"sc-bdVaJa hyzvNA\\">02</td><td data-testid=\\"day-3\\" data-fadate=\\"1397/10/03\\" class=\\"sc-bdVaJa hyzvNA\\">03</td><td data-testid=\\"day-4\\" data-fadate=\\"1397/10/04\\" class=\\"sc-bdVaJa hyzvNA\\">04</td><td data-testid=\\"day-5\\" data-fadate=\\"1397/10/05\\" class=\\"sc-bdVaJa hyzvNA\\">05</td><td data-testid=\\"day-6\\" data-fadate=\\"1397/10/06\\" class=\\"sc-bdVaJa hyzvNA\\">06</td><td data-testid=\\"day-7\\" data-fadate=\\"1397/10/07\\" class=\\"sc-bdVaJa hyzvNA\\">07</td><td data-testid=\\"day-8\\" data-fadate=\\"1397/10/08\\" class=\\"sc-bdVaJa hyzvNA\\">08</td><td data-testid=\\"day-9\\" data-fadate=\\"1397/10/09\\" class=\\"sc-bdVaJa hyzvNA\\">09</td><td data-testid=\\"day-10\\" data-fadate=\\"1397/10/10\\" class=\\"sc-bdVaJa hyzvNA\\">10</td><td data-testid=\\"day-11\\" data-fadate=\\"1397/10/11\\" class=\\"sc-bdVaJa hyzvNA\\">11</td><td data-testid=\\"day-12\\" data-fadate=\\"1397/10/12\\" class=\\"sc-bdVaJa hyzvNA\\">12</td><td data-testid=\\"day-13\\" data-fadate=\\"1397/10/13\\" class=\\"sc-bdVaJa hyzvNA\\">13</td><td data-testid=\\"day-14\\" data-fadate=\\"1397/10/14\\" class=\\"sc-bdVaJa hyzvNA\\">14</td><td data-testid=\\"day-15\\" data-fadate=\\"1397/10/15\\" class=\\"sc-bdVaJa hyzvNA\\">15</td><td data-testid=\\"day-16\\" data-fadate=\\"1397/10/16\\" class=\\"sc-bdVaJa hyzvNA\\">16</td><td data-testid=\\"day-17\\" data-fadate=\\"1397/10/17\\" class=\\"sc-bdVaJa hyzvNA\\">17</td><td data-testid=\\"day-18\\" data-fadate=\\"1397/10/18\\" class=\\"sc-bdVaJa hyzvNA\\">18</td><td data-testid=\\"day-19\\" data-fadate=\\"1397/10/19\\" class=\\"sc-bdVaJa hyzvNA\\">19</td><td data-testid=\\"day-20\\" data-fadate=\\"1397/10/20\\" class=\\"sc-bdVaJa hyzvNA\\">20</td><td data-testid=\\"day-21\\" data-fadate=\\"1397/10/21\\" class=\\"sc-bdVaJa hyzvNA\\">21</td><td data-testid=\\"day-22\\" data-fadate=\\"1397/10/22\\" class=\\"sc-bdVaJa hyzvNA\\">22</td><td data-testid=\\"day-23\\" data-fadate=\\"1397/10/23\\" class=\\"sc-bdVaJa hyzvNA\\">23</td><td data-testid=\\"day-24\\" data-fadate=\\"1397/10/24\\" class=\\"sc-bdVaJa hyzvNA\\">24</td><td data-testid=\\"day-25\\" data-fadate=\\"1397/10/25\\" class=\\"sc-bdVaJa hyzvNA\\">25</td><td data-testid=\\"day-26\\" data-fadate=\\"1397/10/26\\" class=\\"sc-bdVaJa hyzvNA\\">26</td><td data-testid=\\"day-27\\" data-fadate=\\"1397/10/27\\" class=\\"sc-bdVaJa hyzvNA\\">27</td><td data-testid=\\"day-28\\" data-fadate=\\"1397/10/28\\" class=\\"sc-bdVaJa hyzvNA\\">28</td><td data-testid=\\"day-29\\" data-fadate=\\"1397/10/29\\" class=\\"sc-bdVaJa hyzvNA\\">29</td><td data-testid=\\"day-30\\" data-fadate=\\"1397/10/30\\" class=\\"sc-bdVaJa hyzvNA\\">30</td>"`;
